core/src/apps/bitcoin/sign_tx/bitcoin.py

```diff
@@ -455,16 +501,22 @@ async def serialize_segwit_input(self, i: int) -> None:
         # STAGE_REQUEST_SEGWIT_INPUT in legacy
         txi = await helpers.request_tx_input(self.tx_req, i, self.coin)
 
-        if not input_is_segwit(txi):
+        if txi.script_type not in common.SEGWIT_INPUT_SCRIPT_TYPES:
             raise wire.ProcessError("Transaction has changed during signing")
         self.tx_info.check_input(txi)
 
-        node = self.keychain.derive(txi.address_n)
-        key_sign_pub = node.public_key()
+        if txi.script_type == InputScriptType.SPENDP2SHWITNESS:
+            node = self.keychain.derive(txi.address_n)
+            key_sign_pub = node.public_key()
+        else:
+            key_sign_pub = b""
```

Comment:
please add comment explaining that the pubkey is not used in this case

Reply:
https://github.com/trezor/trezor-firmware/pull/1891/commits/ab5fa8d6191c9c12f6b5d8666918c1af81f502b1